test/index.js

```diff
@@ -339,17 +339,21 @@ test('address decoding to buffer', async (t) => {
 test('bootstrap address packing/unpacking', async (t) => {
   t.plan(11)
 
-  const expectedV1MainnetAddress = 'DdzFFzCqrhtBwFyaWje9HStKDWNwWBghBDxGTsnaxoPBE4pZg3pvZC1zDyMpbJqZ7XxpVcHoYc5TA8oA8Hc8gJPUY2kAsaNGW6b8KrrU'
-  const expectedV1TestnetAddress = '2RhQhCGqYPDqLvKWTmnNBHFFUSeMTfdfogpmJMWQE6gmn1bSMiL3ji6Dkjb3YX7UdtaeFSHZBQLKJnPmABFDhp3L2hxvFtPCskG3ep8VpxL1gV'
-  const expectedV2MainnetAddress = 'Ae2tdPwUPEZCxt4UV1Uj2AMMRvg5pYPypqZowVptz3GYpK4pkcvn3EjkuNH'
-  const expectedV2TestnetAddress = '2657WMsDfac6kyMx453f1FZTAVTHcNcuF5pTRD16bcRfGBu53LVPREup3xCV9s5fu'
+  const expectedV1MainnetAddress = 'DdzFFzCqrhspzoFuJ7CyjGUzikzfWEz6DmjeYpB6Dt7WDUDYi8Wv4qaJ2YNnVsMJi8p8yTPLfaheT9NpEAwig4dL9sFNa3ynkauwWuym'
```

Comment:
To make it clear to whoever would be investigating the impact of this change, I'd rather keep the existing test fixtures untouched. The purpose of the fixtures here isn't necessarily to capture a valid v1 and v2 derivation path, but to just ensure the function

Did you perhaps get the public keys added in this testcase from some cardano-wallet test? If so, I'd suggest referencing it as it can provide valuable context to the reader

Reply:
>To make it clear to whoever would be investigating the impact of this change, I'd rather keep the existing test fixtures untouched.

I'm afraid that the old expected addresses were incorrect due to an incorrect packing algorithm. I kept the mnemonic and root keys the same. But we can't ignore the fact that the fixed algorithm derives new addresses, so I changed them.

If you mean the adding of additional level public keys, I did it to align with the approach "tests are docs", to make the process more clear.

>Did you perhaps get the public keys added in this testcase from some cardano-wallet test?

No, they were derived using cardano-crypto.  There is a lot of pre-calculated data in this test script, so these public keys are some of them.

In fact, I just aligned this test with the Cardano specs a bit. Nothing more. I can change back the account key from address to root level, but it will confuse those who think that this is the correct address derivation mechanics. And, ultimately, it may lead to someone creating an issue, where you will have to explain that they are using the wrong keys :)

Reply:
> I'm afraid that the old expected addresses were incorrect due to an incorrect packing algorithm. I kept the mnemonic and root keys the same. But we can't ignore the fact that the fixed algorithm derives new addresses, so I changed them.

Good point with the "documentation as docs" - to be clear, by "fixtures" I meant the test inputs, of course the incorrect outputs (wrong testnet addresses) need to be changed. The fixed function should still derive the same mainnet addresses as far as I understood and tried. So I'd suggest keeping the old test cases just to make a link to the old code for whoever would be comparing changes in versions of this lib.

And on top of fixing these existing testcases I'd suggest introducing your testcase as a new separate test. To prevent somebody from misunderstanding the old testcase as an indicative example of deriving the addresses (good point), I'd suggest renaming the old test case to something like "pack bootstrap addresses (inputs to function not matching standards!)" and your newly introduced test case could be named "pack bootstrap addresses (addresses derived by wallet standards)" and leave in a comment also a reference to the mnemonic from which the public keys/addresses were obtained, that way it should provide the full context to the reader.

Reply:
Ok, good idea.